contrib/babelfishpg_tds/src/backend/tds/tdsrpc.c

```diff
@@ -2334,10 +2334,15 @@ static void
 SendCursorResponse(TDSRequestSP req)
 {
 	int			cmd_type = TDS_CMD_UNKNOWN;
-	Portal		portal;
-
 	/* fetch the portal */
-	portal = GetPortalFromCursorHandle(req->cursorHandle, false);
+	Portal		portal = GetPortalFromCursorHandle(req->cursorHandle, false);
```

Comment:
This is internal implementation of `FetchPortalTargetList` only. Since we require plannedstmt, we implemted similar logic so that we can get both plannedstmt and targetlist for our requirement. I dont know how to test this change though for targetList = NIL. Does it make sense to put `assert(targetList != NIL)` here

Reply:
Existing test cases are sufficient to test this